cloudferrylib/os/image/glance_image.py

```diff
@@ -228,7 +232,7 @@ def delete_image(self, image_id):
         self.glance_client.images.delete(image_id)
 
     def get_image_by_id(self, image_id):
-        for glance_image in self.get_image_list():
+        for glance_image in self.get_image_list_without_filtering():
```

Comment:
Please use `self.glance_client.images.get`. It only lack fields when the are not present. For example, when you create image without `container_format`:

    In [15]: c.images.create(name='image2', copy_from='http://download.cirros-cloud.net/0.3.3/cirros-0.3.3-x86_64-disk.img', is_public=True)
    Out[15]: <Image {u'status': u'active', u'created_at': u'2016-03-15T16:49:58', u'name': u'image2', u'deleted': False, u'checksum': u'133eae9fb1c98f45894a4e60d8736619', u'min_ram': 0, u'updated_at': u'2016-03-15T16:50:01', u'id': u'871b6e5d-37f1-43b2-93a9-d59776d8fe3c', u'owner': u'2645709cdf3d4d569dc407f737734827', u'protected': False, u'min_disk': 0, u'is_public': True, 'properties': {}, u'size': 13200896}>
    In [20]: 'container_format' in  c.images.get('871b6e5d-37f1-43b2-93a9-d59776d8fe3c').to_dict()
    Out[20]: False
    In [21]: c.images.create(name='image2', copy_from='http://download.cirros-cloud.net/0.3.3/cirros-0.3.3-x86_64-disk.img', is_public=True, container_format='bare')
    Out[21]: <Image {u'status': u'queued', u'name': u'image2', u'deleted': False, u'container_format': u'bare', u'created_at': u'2016-03-15T16:53:53', u'disk_format': None, u'updated_at': u'2016-03-15T16:53:53', u'properties': {}, u'owner': u'2645709cdf3d4d569dc407f737734827', u'protected': False, u'min_ram': 0, u'checksum': None, u'min_disk': 0, u'is_public': True, u'deleted_at': None, u'id': u'7e377436-098f-4bb9-b7eb-c51723dfe665', u'size': 13200896}>
    In [22]: 'container_format' in  c.images.get('7e377436-098f-4bb9-b7eb-c51723dfe665').to_dict()
    Out[22]: True